src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1201,6 +1249,22 @@
   {:pre [(kitchensink/datetime? time)]}
   (jdbc/delete! :reports ["producer_timestamp < ?" (to-timestamp time)]))
 
+(defn delete-old-unassociated-catalogs!
+  [time]
+  {:pre [(kitchensink/datetime? time)]}
+  (when @store-catalogs-historically?
+    (let [time (to-timestamp time)]
+      (jdbc/do-prepared
+       (str "DELETE FROM catalogs oc "
+            "       WHERE oc.producer_timestamp < ? "
+            "       AND NOT EXISTS (SELECT 1 FROM catalogs c "
+            "                       JOIN reports r ON c.catalog_uuid = r.catalog_uuid "
+            "                       WHERE c.producer_timestamp < ? AND oc.id = c.id)"
+            "       AND NOT EXISTS (SELECT 1 from catalogs c "
+            "                       JOIN latest_catalogs lc ON c.id = lc.catalog_id "
+            "                       WHERE c.id = oc.id AND c.producer_timestamp < ?)")
```

Comment:
should this cascade to or otherwise handle catalog_resources? without the old fk constraint I don't think it will anymore.